torch/csrc/jit/pickler.cpp

```diff
@@ -218,18 +219,23 @@ void Pickler::pushIValue(const IValue& ivalue) {
     }
   }
   pushIValueImpl(ivalue);
-  if (ivalue.isPtrType()) {
+  if (ivalue.isPtrType() && !ivalue.isString() && !ivalue.isTuple()) {
     memoized_ivalues_.push_back(ivalue);
     memoized_ivalue_map_[ivalue.internalToPointer()] = pushNextBinPut();
   }
 }
 
 void Pickler::pushInt(const IValue& ivalue) {
   auto n = ivalue.toInt();
-  if (n >= std::numeric_limits<int8_t>::min() &&
-      n <= std::numeric_limits<int8_t>::max()) {
+  if (n >= std::numeric_limits<uint8_t>::min() &&
+      n <= std::numeric_limits<uint8_t>::max()) {
```

Comment:
Why this change?

Reply:
Because the old version was incorrect.  `BININT1` can only encode positive values: https://github.com/python/cpython/blob/master/Lib/pickletools.py#L1196